core/sql/generator/GenPreCode.cpp

```diff
@@ -11274,6 +11274,485 @@ short HbaseAccess::extractHbaseFilterPreds(Generator * generator,
+	      (isEncodingNeededForSerialization(colVID.getItemExpr())))
+	    {
+	      castValue = new(generator->wHeap()) CompEncode
+		(castValue, FALSE, -1, CollationInfo::Sort, TRUE, FALSE);
+	    }
+
+	  castValue = castValue->bindNode(generator->getBindWA());
+	  castValue = castValue->preCodeGen(generator);
+
+	  valueVID = castValue->getValueId();
+
+	  NAString nullType;
+
+	  if ((colType.supportsSQLnull()) ||
+	      (valueType.supportsSQLnull()))
+	    {
+	      nullType = "_NULL";
+	    }
+	  else
+	    {
+	      nullType = "";
+	    }
+
+	  // append -NULL to the operator to signify the java code generating pushdown filters to handle NULL semantic logic
+	  if (ie->getOperatorType() == ITM_EQUAL)
+		  op = "EQUAL"+nullType;
+	  else  if (ie->getOperatorType() == ITM_NOT_EQUAL)
+	    op = "NOT_EQUAL"+nullType;
+	  else  if (ie->getOperatorType() == ITM_LESS){
+		  if (flipOp)
+			  op = "GREATER"+nullType;
+		  else
+			  op = "LESS"+nullType;
+	  }
+	  else  if (ie->getOperatorType() == ITM_LESS_EQ){
+		  if (flipOp)
+			  op = "GREATER_OR_EQUAL"+nullType;
+		  else
+			  op = "LESS_OR_EQUAL"+nullType;
+	  }else  if (ie->getOperatorType() == ITM_GREATER){
+		  if (flipOp)
+			  op = "LESS"+nullType;
+		  else
+			  op = "GREATER"+nullType;
+	  }else  if (ie->getOperatorType() == ITM_GREATER_EQ){
+		  if (flipOp)
+			  op = "LESS_OR_EQUAL"+nullType;
+		  else
+			  op = "GREATER_OR_EQUAL"+nullType;
+	  }else
+	    op = "NO_OP"+nullType;
+	}
+    }
+  if (foundUnary){
+	  const NAType &colType = colVID.getType();
+	  NAString nullType;
+
+	  if (colType.supportsSQLnull())
+	    {
+	      nullType = "_NULL";
+	    }
+	  else
+	    {
+	      nullType = "";
+	    }
+	  if (ie->getOperatorType() == ITM_IS_NULL)
+	  		  op = "IS_NULL"+nullType;
+	  else if (ie->getOperatorType() == ITM_IS_NOT_NULL)
+		      op = "IS_NOT_NULL"+nullType;
+  }
+
+  return foundBinary || foundUnary;
+}
+short HbaseAccess::extractHbaseFilterPredsVX(Generator * generator,
+		   ValueIdSet &preds, ValueIdSet &newExePreds){
+	//separate the code that should not belong in the recursive function
+	   if (CmpCommon::getDefault(HBASE_FILTER_PREDS) == DF_OFF)
+	    return 0;
+	   // check if initial (version 1) implementation
+	   if (CmpCommon::getDefault(HBASE_FILTER_PREDS) == DF_MINIMUM)
+	    return extractHbaseFilterPreds(generator,preds,newExePreds);
+
+	   // if here, we are DF_MEDIUM
+	   // cannot push preds for aligned format row
+	   NABoolean isAlignedFormat = getTableDesc()->getNATable()->isAlignedFormat(getIndexDesc());
+
+	   if (isAlignedFormat)
+	     return 0;
+	   //recursive function call
+	   opList_.insert("V2");//to instruct the java side that we are dealing with predicate pushdown V2 semantic, add "V2" marker
+	   extractHbaseFilterPredsV2(generator,preds,newExePreds,FALSE,TRUE);
+	   return 0;
+
+}
+
+// return true if successfull push down of node
+NABoolean HbaseAccess::extractHbaseFilterPredsV2(Generator * generator,
```

Comment:
Yes, I have to respond here because this is one of my pet peeves: Making two methods with the same name and different parameter lists is almost always a BAD idea! Doing it like this with a clear indication in the name of what's the difference is much better. Overloading should be used only sparingly, in cases where it really makes sense - IMHO.